### ISSUE TITLE
Overwriting files with same name when downloading to some folders

### ISSUE THREAD

**Issue by BranescuMihai** [mozilla-mobile/android-components#7839]
This is related to a Fenix ticket: https://github.com/mozilla-mobile/fenix/issues/12159

We override files from subfolders when we download them to the main folder. The problem appears on Android 10, because the MediaStore queried subfolders as well when searching for occurrences, even though it should look only in the destination directory.

**Comment by Mugurell** [comment]
Based on @Amejia481's PR comment about the sample-browser being unaffected I investigated a bit what the differences between Fenix and sample-browser are.

Found the real issue to be related to the use of `MediaStore.Downloads` [when running on >=AndroidQ](https://github.com/mozilla-mobile/android-components/blob/59c509410e9ceef960860776db6375a05919ff7d/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L666).
As part of the "scoped storage" changes this would only give access to the files downloaded by the current app.
This is why the issue is not reproducing on sample-browser.

But Fenix uses `requestLegacyExternalStorage="true"` (and I think other clients of AC would also) which means the app would get access to all files, including the ones not owned by it.

An easy fix would be to also check for the legacy mode,
so the
>     if (SDK_INT >= Build.VERSION_CODES.Q) {
>        useFileStreamScopedStorage()
>     } else {
>        ...
>     }
would become
>     if (SDK_INT >= Build.VERSION_CODES.Q && !Environment.isExternalStorageLegacy()) {
>         ...
>     } else {
>         useFileStreamLegacy()
>     }
